src/test/util/setup_common.cpp

```diff
@@ -321,12 +322,12 @@ CBlock TestChainSetup::CreateBlock(const std::vector<CMutableTransaction>& txns,
         }
     }
 
-    // Replace mempool-selected txns with just coinbase plus passed-in txns:
-    block.vtx.resize(1);
+    Assert(block.vtx.size() == 1);
```

Comment:
if DIP3 active, size of vtx can be > 1
May be here should be "vtx.size() >= 1"?

Reply:
The original backport removes an explicit trim that discards everything except the coinbase transaction and replaces it with an assertion, implying that that the sole existence of a coinbase transaction is expected behaviour. 

Though, certain Dash tests rely on this explicit trim (specifically `block_reward_reallocation_tests`) as the test sequence generates four transactions that it then expects `CreateBlock` to trim, as further down the sequence, assertions fail due to the presence of those transactions it assumes will be trimmed.

```
test_dash: test/block_reward_reallocation_tests.cpp:205: void block_reward_reallocation_tests::block_reward_reallocation::test_method(): Assertion `deterministicMNManager->GetListAtChainTip().HasMN(tx.GetHash())' failed.
unknown location(0): fatal error: in "block_reward_reallocation_tests/block_reward_reallocation": signal: SIGABRT (application abort requested)
```

I've therefore retained the explicit trim but kept the assertion to prevent merge conflicts down the road. As to whether `CreateBlock` should account for `block_reward_reallocation_tests` or vice-versa is subject matter of discussion. @UdjinM6, thoughts?